packages/prism-theme-odyssey/src/utils/CustomPropertyInspector.js

```diff
@@ -0,0 +1,122 @@
+
+    this.renderInspector()
+  }
+
+  setCustomProperties(element) {
+    let prop
+    const styles = getComputedStyle(element)
+    const properties = {}
+
+    for (prop in styles) {
+      const name = styles[prop];
+      const value = Storage.getItem(name) || styles.getPropertyValue(name);        
+
+      if (styles.hasOwnProperty(prop) && styles[prop].includes('--')) { 
+        properties[name] = value
+      }
+
+      if (Storage.getItem(name)) {
+        document.documentElement.style.setProperty(name, value);
+      }
+    }
+    return properties
+  }
+
+
+  renderInspector() {
+    const { properties } = this
+    const inspector = document.createElement('form');
+    inspector.classList.add('custom-properties-inspector');
+    inspector.style = `
+      position: sticky;
+      top: 0;
+      border-right: 5px solid #f5f5f6;
+      padding:  3.42857rem 1.71429rem;
+      max-height: 100vh;
+      overflow: scroll;
+      min-width: 16rem;
+    `
+
+    const inspectorTitle = document.createElement('h5');
+    inspectorTitle.innerHTML = "Custom Properties Inspector";
+    inspectorTitle.style="margin-bottom: 2rem;"
+    inspector.appendChild(inspectorTitle)
+
+    const inspectorReset = document.createElement('button');
+    inspectorReset.classList.add('ods-button', 'is-ods-button-danger', 'is-ods-button-full-width');
+    inspectorReset.innerHTML = "Reset Theme";
+
+    document.body.style = "display: flex;"
+
+    for (const [key, value] of Object.entries(properties)) {
+      const fieldset = document.createElement('fieldset');
+      const fieldsetFlex = document.createElement('div');
+      const input = document.createElement('input');
+      const label = document.createElement('label');
+
+      // Set Odyssey Fieldset
+      fieldset.classList.add('ods-fieldset')
+
+      // Set Odyssey Fieldset Flex
+      fieldsetFlex.classList.add('ods-fieldset-flex')
+
+      // Set Odyssey Label
+      label.classList.add('ods-label');
+      label.innerHTML = key;
+      
+      // Set Odyssey Text Input
+      input.setAttribute('type', 'text');
+      input.classList.add('ods-text-input');
+      input.name = key
+      input.value = value.replace(/([A-Z]+)/g, " $1").replace(/([A-Z][a-z])/g, " $1");
+
+      // Append all the elements
+      fieldset.appendChild(fieldsetFlex)
+      fieldsetFlex.appendChild(label);
+      fieldsetFlex.appendChild(input);
+      inspector.appendChild(fieldset);
+      inspector.appendChild(inspectorReset)
+    }
+
+    if (Object.keys(properties).length === 0 && properties.constructor === Object) {
+      const errorNotice = document.createElement('div');
+      errorNotice.innerHTML = `
+        <div>
+          <div style="font-size: 6rem; text-align: center;" aria-label="No good">⚠️</div>
+          <p>The Custom Property Inspector has nothing to show. This is probably because you are <a href="https://github.com/w3c/csswg-drafts/issues/1316">using Google Chrome.</a></p>
```

Comment:
chrome has no way to inspect custom property values through Javascript?

Reply:
Not at the moment. This is a bug within chrome, which will be fixed at some point in the future.

Chromium Issue: https://bugs.chromium.org/p/chromium/issues/detail?id=949807